docs/07_how-to/21_web-unit-test.mdx

```diff
@@ -0,0 +1,235 @@
+import {Connect, Message} from "@genesislcap/foundation-comms";
+import {DOM} from "@genesislcap/web-core";
+import {ListItems} from "../../src/routes/list-items/list-items";
+
+ListItems; // reference to avoid tree shaking.
+
+const rowData = [...];
+
+const mockData = {
+  "ROWS_COUNT": 100,
+  "MESSAGE_TYPE": "QUERY_UPDATE",
+  "ROW": rowData,
+  "MORE_ROWS": false,
+  "SOURCE_REF": "f5ebbfcd-7753-41d0-a1ec-eda8ffc307a3",
+  "SEQUENCE_ID": 1
+}
+
+// partial mock of the Connect service so that snapshots return our mock data during these tests
+const connectMock = {
+  snapshot: (): Promise<Message> => Promise.resolve(mockData)
+};
+
+const mocks = [
+  Registration.instance(Connect, connectMock),
+];
+
+const Suite = createComponentSuite<ListItems>('ListItems', 'list-items-container', undefined, mocks);
+
+Suite('Can be created in the DOM.', async ({element}) => {
+  assert.ok(element);
+});
+
+Suite('Displays a list item for every row.', async ({element}) => {
+  await DOM.nextUpdate();
+  const content = element.shadowRoot?.querySelector('div.content');
+  const listItems = content.querySelectorAll('div.list-item');
+  assert.equal(listItems.length, element.entities.length);
+});
+
+Suite('Can navigate to details from a list item', async ({element}) => {
+  await DOM.nextUpdate();
+  const content = element.shadowRoot?.querySelector('div.content');
+  const firstListItem = content.querySelectorAll('div.list-item')[0];
+  const anchor = firstListItem.querySelector('rapid-anchor');
+  assert.ok(anchor);
+  await (anchor as HTMLElement)?.click();
+  assert.equal(window?.location?.pathname, `/list-item-detail/${element.entities[0].TRADE_ID}`);
+});
+
+Suite.run();
+```
+
+### Testing logic
+
+You can use the `createLogicSuite` utility provided by `@genesislcap/foundation-testing` to test function output based on specific input arguments.
+
+Below is an example file **utils.test.ts**, which tests that our `timeOfDay` utility function returns the correct time of day for any input hour. It returns `null` for invalid input arguments.
+
+```typescript
+import {createLogicSuite} from "@genesislcap/foundation-testing";
+import {timeOfDay} from "../../src/utils/util";
+
+const timeOfDaySuite = createLogicSuite('timeOfDay');
+
+timeOfDaySuite('timeOfDay should return the correct time of day', ({runCases}) => {
+  runCases(timeOfDay, [
+    [[0], 'night'],
+    [[5], 'night'],
+    [[6], 'morning'],
+    [[11], 'morning'],
+    [[12], 'afternoon'],
+    [[16], 'afternoon'],
+    [[17], 'evening'],
+    [[20], 'evening'],
+    [[21], 'night'],
+    [[24], 'night'],
+    [[25], null],
+    [[-1], null],
+    [[false], null],
+    [['banana'], null],
+    [[{foo: 'bar'}], null],
+    [[[1, 2, 3]], null],
+  ]);
+});
+timeOfDaySuite.run();
+```
+
+## Unit testing scripts in-depth
+
+Below is a comprehensive guide on how to use these scripts to run your unit tests effectively, along with explanations of each script.
```

Comment:
Using "comprehensive" and "in-depth" (title above) seems a bit odd when we have minimal detail below